zipline/testing/core.py

```diff
@@ -1470,6 +1475,188 @@ def ensure_doctest(f, name=None):
     return f
 
 
+def run_correlations(pearson_factor,
```

Comment:
The two things using this testing logic is 1) the built-in factors `RollingPearsonOfReturns`, `RollingLinear...` etc. and 2) the `Factor` methods `pearsonr`, `spearmanr` and `linear_regression` when passed two 2D factors. In both (1) and (2) we want to compute the expected results by running a `Returns` factor and calculating correlations/regressions over the `Returns` results. While cases (1) and (2) each definitely need their own test, after thinking more about it, I think that (1) and (2) are disjoint enough that attempting to write a shared function here is too convoluted. The best option is probably for each test to just re-write this logic, even though that makes for rather long tests, but there's no way to get around that anyway.